Initial work for setting up opentelemetry package (#6131)
Azure/azure-sdk-for-java: sdk/core/azure-core-tracing-opentelemetry/README.md

# Azure Tracing client library for Java

> This package enables distributed tracing across Azure SDK Java libraries through [opentelemetry][opentelemetry]. OpenTelemetry is an open source, vendor-agnostic, single distribution of libraries to provide metrics collection and distributed tracing for services. 

Reviewer: ```suggestion
This package enables distributed tracing across Azure SDK Java libraries through [opentelemetry][OpenTelemetry]. OpenTelemetry is an open source, vendor-agnostic, single distribution of libraries to provide metrics collection and distributed tracing for services. 
```

Reviewer: Whichever is the shown text should use the correct capitalization for the product.

Author: updated to use `OpenTelemetry`

> The Azure core tracing package provides:

Reviewer: ```suggestion
The Azure Core OpenTelemetry tracing package provides:
```

Reviewer: Just in case we continue to add additional tracing packages in the future.

> - Context propagation, in order to correlate activities and requests between services with an initial customer action.

Reviewer: ```suggestion
- Context propagation, used to correlate activities and requests between services with an initial customer action.
```

- Tracing user requests to the system, allowing to pinpoint failures and performance issues.

[Source code][source_code] | [API reference documentation][api_documentation] | [Product
documentation][api_documentation] | [Samples][samples]

## Getting started
### Prerequisites
- Java Development Kit (JDK) with version 8 or above
- [Maven][maven]

### Adding package to your product
[//]: # ({x-version-update-start;com.azure:azure-core-tracing-opentelemetry;current})
```xml
<dependency>
  <groupId>com.azure</groupId>
  <artifactId>azure-core-tracing-opentelemetry</artifactId>
  <version>1.0.0-preview.1</version>
</dependency>
```
[//]: # ({x-version-update-end})

> ### Default HTTP Client

Reviewer: Does OpenTelemetry take a dependency on an HTTP client? If not we should remove this section.

Author: As discussed offline, users might need to know about this when using this plugin in conjunction with any of our client libraries, so keeping this.

### Span
A span represents a single operation in a trace. A span could be representative of an HTTP request, a remote procedure call (RPC), a database query, or even the path that a code takes.

## Examples
The following sections provides examples of using the azure-core-tracing-opentelemetry plugin with some of the Azure Java SDK libraries:
### Using the plugin package with HTTP client libraries
- Synchronously create a secret using [azure-security-keyvault-secrets][azure-security-keyvault-secrets] with tracing enabled.
    
    Users can additionally pass the value of the current tracing span to the SDKs using key **PARENT_SPAN_KEY** on the [Context][context] parameter of the calling method.
    The plugin package creates a root span to encapsulate all the child spans created in the calling methods when no parent span is passed in the context.
    This [sample][sample_key_vault] provides an example when no user parent span is passed.
    ```java
    private static  final Tracer TRACER;
    private static final TracerSdkFactory TRACER_SDK_FACTORY;
        
        static {
            TRACER_SDK_FACTORY = configureOpenTelemetryAndJaegerExporter();

>             TRACER = TRACER_SDK_FACTORY.get("Azure-OpenTelemetry");

Reviewer 2: ```suggestion
            TRACER = TRACER_SDK_FACTORY.get("Sample");
```

        }

        public static void doClientWork() {
          SecretClient client = new SecretClientBuilder()
            .endpoint("<your-vault-url>")
            .credential(new DefaultAzureCredentialBuilder().build())
            .buildClient();
                    
          Span span = TRACER.spanBuilder("user-parent-span").startSpan();
          try (Scope scope = TRACER.withSpan(span)) {
              final Context traceContext = new Context(PARENT_SPAN_KEY, span);
              secretClient.setSecretWithResponse(new Secret("secret_name", "secret_value", traceContext));
          } finally {
              span.end();
          }
        }
    ```

### Using the plugin package with AMQP client libraries
Send a single event/message using [azure-messaging-eventhubs][azure-messaging-eventhubs] with tracing enabled.
    
Users can additionally pass the value of the current tracing span to the EventData object with key **PARENT_SPAN_KEY** on the [Context][context] object:

```java
private static final Tracer TRACER;
private static final TracerSdkFactory TRACER_SDK_FACTORY;
    
    static {
        TRACER_SDK_FACTORY = configureOpenTelemetryAndJaegerExporter();

>         TRACER = TRACER_SDK_FACTORY.get("Azure-OpenTelemetry");

Reviewer 2: ```suggestion
        TRACER = TRACER_SDK_FACTORY.get("Sample");
```
